models/spring-ai-bedrock-converse/src/main/java/org/springframework/ai/bedrock/converse/BedrockProxyChatModel.java

```diff
@@ -0,0 +1,710 @@
+ * <p>
+ * https://docs.aws.amazon.com/bedrock/latest/userguide/model-ids.html
+ * <p>
+ * https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters.html
+ *
+ * @author Christian Tzolov
+ * @author Wei Jiang
+ * @since 1.0.0
+ */
+public class BedrockProxyChatModel extends AbstractToolCallSupport implements ChatModel {
+
+	private static final Logger logger = LoggerFactory.getLogger(BedrockProxyChatModel.class);
+
+	private static final ChatModelObservationConvention DEFAULT_OBSERVATION_CONVENTION = new DefaultChatModelObservationConvention();
+
+	private final BedrockRuntimeClient bedrockRuntimeClient;
+
+	private final BedrockRuntimeAsyncClient bedrockRuntimeAsyncClient;
+
+	private FunctionCallingOptions defaultOptions;
+
+	/**
+	 * Observation registry used for instrumentation.
+	 */
+	private final ObservationRegistry observationRegistry;
+
+	/**
+	 * Conventions to use for generating observations.
+	 */
+	private ChatModelObservationConvention observationConvention;
+
+	public BedrockProxyChatModel(BedrockRuntimeClient bedrockRuntimeClient,
+			BedrockRuntimeAsyncClient bedrockRuntimeAsyncClient, FunctionCallingOptions defaultOptions,
+			FunctionCallbackContext functionCallbackContext, List<FunctionCallback> toolFunctionCallbacks,
+			ObservationRegistry observationRegistry) {
+
+		super(functionCallbackContext, defaultOptions, toolFunctionCallbacks);
+
+		Assert.notNull(bedrockRuntimeClient, "bedrockRuntimeClient must not be null");
+		Assert.notNull(bedrockRuntimeAsyncClient, "bedrockRuntimeAsyncClient must not be null");
+
+		this.bedrockRuntimeClient = bedrockRuntimeClient;
+		this.bedrockRuntimeAsyncClient = bedrockRuntimeAsyncClient;
+		this.defaultOptions = defaultOptions;
+		this.observationRegistry = observationRegistry;
+	}
+
+	/**
+	 * Invoke the model and return the response.
+	 *
+	 * https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters.html
+	 * https://docs.aws.amazon.com/bedrock/latest/APIReference/API_runtime_Converse.html
+	 * https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/bedrockruntime/BedrockRuntimeClient.html#converse
+	 * @param bedrockConverseRequest Model invocation request.
+	 * @return The model invocation response.
+	 */
+	@Override
+	public ChatResponse call(Prompt prompt) {
+
+		ConverseRequest converseRequest = this.createRequest(prompt);
+
+		ChatModelObservationContext observationContext = ChatModelObservationContext.builder()
+			.prompt(prompt)
+			.provider(AiProvider.BEDROCK_CONVERSE.value())
+			.requestOptions(buildRequestOptions(converseRequest))
+			.build();
+
+		ChatResponse chatResponse = ChatModelObservationDocumentation.CHAT_MODEL_OPERATION
+			.observation(this.observationConvention, DEFAULT_OBSERVATION_CONVENTION, () -> observationContext,
+					this.observationRegistry)
+			.observe(() -> {
+
+				ConverseResponse converseResponse = this.bedrockRuntimeClient.converse(converseRequest);
```

Comment:
Should we add a retry mechanism via `RetryTemplate` in case something might happen?

Reply:
I've noticed that the AWS SDK has a built-in retry utilities. 
But we need to expose and configure them hopefully they will support the async model too.

Reply:
https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/retry-strategy.html